### DISCUSSION TITLE
0609 피어세션일지

### DISCUSSION THREAD

**Discussion by gvsteve24** [bcaitech1/p4-dkt-freshtomato#103] [General]
# 오늘의 TMI + 이모지 

| 재훈        | 민용  | 유라        | 아라            | 정현       | 진현      |
| ----------- | ----- | ----------- | --------------- | ---------- | --------- |
| :male_mage: | :bed: | :honey_pot: | :drooling_face: | :moneybag: | :cyclone: |

# 회의 및 토론

### 실험 진행 상황 

**[민용]**

**jinhyun feature _ 적용**

fold별로 확오르는 것도 있었고 아닌것도 있었다.

**(난이도 기존해석 + ) 진현피처 or 진현피처** 
거의 떨어지거나 비슷하다 // valid는 조금 올랐는데 LB는 떨어짐 ( LB .813 )

pseudo labeling을 해봄 
직접 짰는데 daily mission에 있는거랑 비교해봐야겠다.
학습한 결과를 또 반복하고 또 반복하고 ! 해서 개선해보는 것은 어떨지

pretraining을 모델 백본만 쓰려고 (나중에 불러오려고)

​	각각 파라미터를 층으로 저장해서 load_state_state (예를들면 GRU+Encoder)
​	GRU 부분이랑 ATT 따로 저장해서 모델 선언할때 불러와서 쓰려고 함.

**[hyperparameter 추천]** 

SAINT-b128-lay1-enlay4-seq250-hid512-seed42-lr1e5-drop_0.2-opt_adamW-sch_CosAnWarmv2-pat10-kfold10-cat2-cont3-emb_3 

교육학적관점이나 .... 유의미한 실험 (실력 평가하는 지표..가 뭘까.... 검색해보면 좋을거같다. )

**[유라]**

custom kfold 속도저하가 안나아짐 => ~~재훈꺼를 기다려볼 예정~~ 개선성공!

테스트셋이랑 같은 문제를 푼것만 validation을 만들고 하는데

(validation filter를 안걸고) 전체를 가지고 하는 것으로 하는데 0.81정도 나오고 0.79 이렇게 떨어졌음

**[아라]**

기존 피처에 pretraining => .73

pretrain은 parameter 의 키에서 불러오려고 했음 

cate_cols = ["problem_number", "nth_test", ~~"time_elapsed_cat"~~]
cont_cols = ["assessmentItemID_mean", "KnowledgeTag_mean", "testId_mean", ~~"time_elapsed", "user_problem_cumcount"~~]

tag_mean을 제외하면 valid 기준 성능이 향상됨 

여러 feature를 넣어봤는데 떨어짐.... 대부분 떨어져서, 다른걸 추가하기가 어려워짐

피처가 넣어줄수록 너무 떨어지는 경향

민용) 피처가 잘못됬다기 보다 이미 겹치는 의미의 피처들일 수 있다라는 생각

​	last query 에 맞는 애가 뭔지 알아봐야될 것같다

​	layer ? 4, max_length = 250를 좀 늘려도 괜찮을거같다는 생각

**[재훈]**   time diff 해서 구간 나눠주는 것 스스로 짜봄 + pd.cut

```python
## (비교) 민용 코드
def time_for_cat(x):
        for i in range(1, 30):
            if x < 10 * i:
                return i
        return 30

df["time_elapsed_cat"] = df.time_elapsed.apply(time_for_cat)
```

진현 피처 쓰고 nan 이 나오던 부분이 있었음 => save_dict를 쓰고 나중에 가져오는 걸로 했음

lightgbm 은 시간을 줄여봤다. (1fold에 5분소요 >> 6초)

random으로 valid index를 썼더니 valid개수가 증가하게 됨
마지막 유저로 맞춰줌 .... > 수정되어 valid개수가 일정해짐

positional indexers are out of bounds  에러가 뜸
x, y, groups => indexiable => np.array를 써서 (base kfold )+ 상속 + 

**permutation importance를 이용할 예정** 

~~create_model (cross_validation = False) 하고 넘어감~~ 개선!

validation 42, 43 으로 작았는데  결과가 괜찮았다(testset의 대표성을 반영). 

**[진현]**   실험 조건 / 2 layer, 2 head,  max_seq = 100, epoch = 30

**SAINT에 들어가는게 논문이랑 조금 다르게 구현되어있는거 같다.**

현재는 encoder에 들어갈 임베딩 + decoder에 들어갈 임베딩 인데

기존 트랜스포머를 (nn.transformer) decoder, encoder를 가져와서 넣어주는데 

근데 안에 구조 수정을 어떻게 해야될지 모르겠다.

민용) upper triangle 이 있는데 여기도 있네, 논문에서는 upper triangle mask를 해주긴 하거든..? 고려해보기

decoder마스크만 적용하는 거 고려해보기

**[정현]**

LSTM ATT 재훈 데이터로 계속 실험할 예정

### 향후 실험 계획

**앙상블 시도는 언제부터 해볼꺼?**
제출 파일들로 합쳐볼건데 3, 4개 정도는 앙상블로 조합해보자
**Discusssion에 csv파일을 하나씩 모아두는 거 어떻게 생각함?** 

**11일 멘토링까지는 꼭해봐야될 것들**
앙상블은 꼭 해서 말씀드려야됨!
( 디스커션 = csv, 어떤 모델썼었는지 , 단일모델에서는 성능이 어땠는지 까지 올려주기 )
모델 서빙 강의 듣고 해보길 바람

[민용] max_length_seq = 1800(max) Saint는 올려서 올려서 해보기

정답률 결합한 피처로 테스트해보고 공유하기

# 저녁메뉴 공유공유

| 재훈            | 민용          | 유라            | 아라          | 정현            | 진현          |
| --------------- | ------------- | --------------- | ------------- | --------------- | ------------- |
| 🍣| :ddeok-bokki: | :grey_question: | :green_salad: | :motor_scooter: | :green_salad: |

**Comment by yukyunglee** [comment]
너무 조급해하지마시고 가능한 capa 내에서 파이팅입니다 !!
